bin/taiyi-challenger/src/main.rs

```diff
@@ -0,0 +1,951 @@
+use std::{collections::HashSet, str::FromStr, sync::Arc};
+
+use alloy_eips::{eip2718::Encodable2718, merge::SLOT_DURATION_SECS, BlockNumberOrTag};
+use alloy_primitives::{hex, Address, Bytes, U256};
+use alloy_provider::{Provider, ProviderBuilder, WsConnect};
+use alloy_signer::k256::{self};
+use alloy_sol_types::sol;
+use clap::Parser;
+use futures_util::{future::join_all, StreamExt};
+use redb::{Database, TableDefinition};
+use reqwest::Url;
+use reqwest_eventsource::{Event, EventSource};
+use taiyi_primitives::{
+    PreconfRequest, PreconfRequestTypeA, PreconfRequestTypeB, PreconfResponseData,
+};
+use tracing::{debug, error, level_filters::LevelFilter};
+
+mod preconf_request_data;
+use preconf_request_data::{Bincode, PreconfRequestData};
+
+const PRECONF_TABLE: TableDefinition<u64, Bincode<Vec<PreconfRequestData>>> =
+    TableDefinition::new("preconf");
+const CHALLENGE_TABLE: TableDefinition<u64, Bincode<Vec<PreconfRequestData>>> =
+    TableDefinition::new("challenge");
+
+pub fn get_slot_from_timestamp(timestamp: u64, genesis_timestamp: u64) -> u64 {
+    (timestamp - genesis_timestamp) / SLOT_DURATION_SECS
+}
+
+#[derive(Parser, Clone)]
+struct Opts {
+    /// execution_client_ws_url
+    #[clap(long = "execution-client-ws-url")]
+    execution_client_ws_url: String,
+    /// beacon_url
+    #[clap(long = "beacon-url")]
+    beacon_url: String,
```

Comment:
```suggestion
    #[clap(long = "network")]
    network: String,
```

I think `becaon_url` is only used to get the genesis time so its safe to replace it with the `network` argument.
Now genesis_time can be calculate as: 
```
use ethereum_consensus::{deneb::Context, networks::Network};

let network: Network = opts.network.clone().into();
let context: Context = network.try_into()?;
let genesis_time = context.genesis_time()?;
```